code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -593,15 +593,22 @@
 	if(candidates.len)
 		theghost = pick(candidates)
 		var/mob/living/simple_animal/hostile/guardian/G = spawn_guardian(user, theghost.key)
-		var/time_limit = world.time + 300
-		var/guardianNewName = stripped_input(G.summoner, "You are the user of [G.name]. Would you like to name your guardian something else?", "Name Guardian", G.name, MAX_NAME_LEN)
-		if(world.time < time_limit)
-			if(length(guardianNewName) > 0)
+		//Give the stand user 3 chances to rename their stand
+		for(var/i = 2, i >= 0,i--)
+			var/guardianNewName = stripped_input(user, "You are the user of [G.name]. Would you like to name your guardian something else?", "Name Guardian", G.name, MAX_NAME_LEN)
+			guardianNewName = reject_bad_name(guardianNewName, 1)
+			if(!isnull(guardianNewName))
+				//Valid name, set it and break loop
 				G.name = guardianNewName
+				return
 			else
-				user << "<span class='danger'>Sorry, you can't have a null name!</span>"
-		else
-			user << "<span class='danger'>Sorry, you didn't give a new name in time!</span>"
+				if(i > 1)
+					user << "<span class='danger'>That's an invalid name! You have [i] more attempts.</span>"
```

Comment:
instead of 2 ifs for i > 1 and i == 0, you could do if(i)  user << "<span class='danger'>That's an invalid name! You have [i] more [i > 1 ? "attempts" : "attempt"].</span>"